### PR TITLE
Update usage of list and call getItems when needed

### LINKED ISSUE #411: MigrateItemWriterWrite should update invocations of list to `Chunk.getItems()`
https://github.com/openrewrite/rewrite-spring/blob/f9fd034692c5121ea36de006638274fad0a05a92/src/main/java/org/openrewrite/java/spring/batch/MigrateItemWriterWrite.java#L32

Will correctly change the method declaration which has a `List` parameter to instead be a `Chunk`. 

Throughout the method body invocations of `List` methods also need to be updated to call `Chunk.getItems()` first

### PR DESCRIPTION
**Pull request by joanvr**
## What's changed?
After changing the type of the method, we can have scenarios where the method was manipulating the original List, and the used methods does not exist in Chunk. So we need to call `Chunk::getItems` to retrieve the actual list and keep compilable code.
Also calling getItems when needed on assignments, variable initialization and method invocations (if the destination type is not `java.lang.Iterable` since it's the only common interface between List and Chunk).

## What's your motivation?
Fixes #411 

## Anything in particular you'd like reviewers to focus on?
Any other corner case where we should call getItems?

## Anyone you would like to review specifically?
@timtebeek @sambsnyd 

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've added the license header to any new files through `./gradlew licenseFormat`
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/main/java/org/openrewrite/java/spring/batch/MigrateItemWriterWrite.java [hunk 1]
```diff
@@ -102,4 +107,105 @@ public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, Ex
...
+            return a;
+        }
+
+        private boolean notAssignableFromChunk(J.VariableDeclarations.NamedVariable var) {
+            return var.getVariableType() != null && notAssignableFromChunk(var.getVariableType().getType());
+        }
+
+        private boolean notAssignableFromChunk(@Nullable JavaType type) {
+            // Iterable is the only common type between List and Chunk
+            return type != null && !type.toString().startsWith(ITERABLE_FQN);
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/spring/batch/MigrateItemWriterWrite.java:185]: Shouldn't we prefer TypeUtils over String comparisons here?
```suggestion
            return type != null && !TypeUtils.isOfClassType(type, ITERABLE_FQN);
```

File: src/main/java/org/openrewrite/java/spring/batch/MigrateItemWriterWrite.java [hunk 2]
```diff
@@ -102,4 +107,105 @@ public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, Ex
...
+        }
+
+        private boolean isParameter(J.Identifier maybeParameter) {
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/spring/batch/MigrateItemWriterWrite.java:188]: Rather than doing four casts from each call above, wouldn't it be easier to pull the J.Identifier type check in here?

File: src/main/java/org/openrewrite/java/spring/batch/MigrateItemWriterWrite.java [hunk 3]
```diff
@@ -102,4 +107,105 @@ public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, Ex
...
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
+            J.MethodInvocation mi = super.visitMethodInvocation(method, executionContext);
+            if (LIST_MATCHER.matches(mi) && mi.getSelect() instanceof J.Identifier) {
+                if (isParameter((J.Identifier) mi.getSelect())) {
+                    assert mi.getPadding().getSelect() != null;
+                    // No need to take care of typing here, since it's going to be printed and parsed on the JavaTemplate later on.
+                    mi = mi.withSelect(newGetItemsMethodInvocation(mi.getPadding().getSelect()));
+                }
+            }
+            if (mi.getMethodType() != null && !ITEM_WRITER_MATCHER.matches(mi)) {
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/spring/batch/MigrateItemWriterWrite.java:132]: Matcher already does that null check.
```suggestion
            if (!ITEM_WRITER_MATCHER.matches(mi)) {
```

**Review by timtebeek — APPROVED** [review]
A few small comments on my end; otherwise approved. Hard to say if there's any area's we haven't covered yet.

File: src/main/java/org/openrewrite/java/spring/batch/MigrateItemWriterWrite.java (hunk 1 above)
Comment (timtebeek) [reply src/main/java/org/openrewrite/java/spring/batch/MigrateItemWriterWrite.java:185]: Might need an import if applied through GitHub.

**Review by timtebeek — APPROVED** [review]
